### PR TITLE
PT-443 es orders controllers

### PR DESCRIPTION
**Pull request by atapin**

### CODE REVIEW

File: core/src/main/kotlin/com/rarible/protocol/union/core/model/EsOrderFilter.kt [hunk 1]
```diff
@@ -82,4 +84,166 @@ data class EsOrderByIdsFilter(
...
+
+            if(origin != null) {
+                add(QueryBuilders.termsQuery(EsOrder::origins.name, origin))
+            }
+
+        }
+
+        return genericBuild(continuation, size, sort, *list.toTypedArray())
+    }
+
```
Comment (augur) [inline core/src/main/kotlin/com/rarible/protocol/union/core/model/EsOrderFilter.kt:248]: Lets add integration tests on these filters, as in other entities

**Review by augur — CHANGES_REQUESTED** [review]

**Comment by augur** [comment]
should be pointed to master, I believe

File: core/src/main/kotlin/com/rarible/protocol/union/core/model/EsOrderFilter.kt [hunk 2]
```diff
@@ -82,4 +84,166 @@ data class EsOrderByIdsFilter(
...
+            QueryBuilders.termsQuery(
+                "make.address",
+                itemId
+            )
+            QueryBuilders.termsQuery(
+                EsOrder::type.name,
+                EsOrder.Type.SELL.name
+            )
+
+            if(platform != null) {
```
[the remark this answers is not in the document]
Comment (atapin) [reply core/src/main/kotlin/com/rarible/protocol/union/core/model/EsOrderFilter.kt:111]: currently it's not possible to reuse the util function, as it's located in enrichment module. let's stay with builders approach for now

File: core/src/main/kotlin/com/rarible/protocol/union/core/model/EsOrderFilter.kt (hunk 1 above)
Comment (atapin) [reply core/src/main/kotlin/com/rarible/protocol/union/core/model/EsOrderFilter.kt:234]: done

File: api/src/main/kotlin/com/rarible/protocol/union/api/service/elastic/OrderElasticService.kt [hunk 3]
```diff
@@ -138,6 +165,40 @@ class OrderElasticService(
...
+
+    private suspend fun fetchOrders(orderFilter: EsOrderFilter): OrdersDto {
+        val esOrders = esOrderRepository.findByFilter(orderFilter)
+        val orderIdsByBlockchain = esOrders.groupBy(EsOrder::blockchain, EsOrder::orderId)
+
+        val slices: List<OrderDto> = orderIdsByBlockchain.flatMapAsync { (blockchain, ids) ->
+            router.getService(blockchain).getOrdersByIds(ids)
+        }
+
+        return if(esOrders.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (atapin) [reply api/src/main/kotlin/com/rarible/protocol/union/api/service/elastic/OrderElasticService.kt:191]: what do you mean?

File: api/src/main/kotlin/com/rarible/protocol/union/api/service/elastic/OrderElasticService.kt [hunk 4]
```diff
@@ -138,6 +165,40 @@ class OrderElasticService(
...
+            OrdersDto(
+                orders = slices,
+                continuation = DateIdContinuation(
+                    id = last.orderId,
+                    date = last.lastUpdatedAt
+                ).toString()
+            )
+        }
     }
 }
```
[the remark this answers is not in the document]
Comment (atapin) [reply api/src/main/kotlin/com/rarible/protocol/union/api/service/elastic/OrderElasticService.kt:204]: repository integration tests should cover the main logic

[the remark this answers is not in the document]
Comment (augur) [reply api/src/main/kotlin/com/rarible/protocol/union/api/service/elastic/OrderElasticService.kt:204]: Meanwhile, repository tests:
https://github.com/rarible/union-service/blob/0ab261240df885e10593d24753cd3cf0e1ea2390/indexer/src/test/kotlin/com/rarible/protocol/union/search/indexer/repository/EsOrderRepositoryFt.kt
Maybe you're talking about some others?

File: api/src/main/kotlin/com/rarible/protocol/union/api/service/elastic/OrderElasticService.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (augur) [reply api/src/main/kotlin/com/rarible/protocol/union/api/service/elastic/OrderElasticService.kt:191]: I mean that orders `esOrders` are returned from ES in specific sorting, but during `orderIdsByBlockchain.flatMapAsync` this sorting breaks, whereas it has to be maintained

File: api/src/main/kotlin/com/rarible/protocol/union/api/service/elastic/OrderElasticService.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (atapin) [reply api/src/main/kotlin/com/rarible/protocol/union/api/service/elastic/OrderElasticService.kt:209]: fixed

File: api/src/main/kotlin/com/rarible/protocol/union/api/service/elastic/OrderElasticService.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (atapin) [reply api/src/main/kotlin/com/rarible/protocol/union/api/service/elastic/OrderElasticService.kt:196]: fixed

File: enrichment/src/main/kotlin/com/rarible/protocol/union/enrichment/repository/search/ElasticSearchRepository.kt [hunk 5]
```diff
@@ -100,6 +113,11 @@ abstract class ElasticSearchRepository<T>(
         ).awaitFirstOrNull()?.deleted
     }
 
+    //for tests
+    suspend fun deleteAll() {
```
Comment (augur) [inline enrichment/src/main/kotlin/com/rarible/protocol/union/enrichment/repository/search/ElasticSearchRepository.kt:117]: It was discussed previously that it is dangerous to have such method, though solely for tests.
How we deal with it for other entities?

Comment (atapin) [reply enrichment/src/main/kotlin/com/rarible/protocol/union/enrichment/repository/search/ElasticSearchRepository.kt:117]: is was supposed to be removed. removed now

File: api/src/main/kotlin/com/rarible/protocol/union/api/service/elastic/OrderElasticService.kt [hunk 6]
```diff
@@ -138,6 +164,46 @@ class OrderElasticService(
...
+
+        val slices = orderIdsByBlockchain.flatMapAsync { (blockchain, ids) ->
+            val isBlockchainEnabled = router.isBlockchainEnabled(blockchain)
+            if(isBlockchainEnabled) {
+                val rawIds = ids.map { IdParser.parseOrderId(it).value }
+                router.getService(blockchain).getOrdersByIds(rawIds)
+            } else emptyList()
+        }.associateBy { it.id.fullId() }
+
+        val sortedOrders = esOrders.map { slices[it.orderId]!! }
```
[the remark this answers is not in the document]
Comment (atapin) [reply api/src/main/kotlin/com/rarible/protocol/union/api/service/elastic/OrderElasticService.kt]: fixed

File: core/src/main/kotlin/com/rarible/protocol/union/core/FeatureFlagsProperties.kt [hunk 7]
```diff
@@ -21,6 +22,7 @@ data class FeatureFlagsProperties(
     // orders
     var enableOrderQueriesToElasticSearch: Boolean = false,
     var enableOrderSaveImmediateToElasticSearch: Boolean = false,
+    var orderRefreshPolicy: WriteRequest.RefreshPolicy = WriteRequest.RefreshPolicy.NONE,
```
[the remark this answers is not in the document]
Comment (atapin) [reply core/src/main/kotlin/com/rarible/protocol/union/core/FeatureFlagsProperties.kt]: fixed

**Review by augur — APPROVED** [review]

**Pull request merged**